lightning/src/ln/channelmanager.rs

```diff
@@ -3321,6 +3383,98 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 			self.finish_force_close_channel(failure);
 		}
 	}
+
+	fn set_payment_hash_secret_map(&self, payment_hash: PaymentHash, payment_preimage: Option<PaymentPreimage>, min_value_msat: Option<u64>, invoice_expiry_delta_secs: u32, user_payment_id: u64) -> Result<PaymentSecret, APIError> {
+		assert!(invoice_expiry_delta_secs <= 60*60*24*365); // Sadly bitcoin timestamps are u32s, so panic before 2106
```

Comment:
Should we also check that the given expiry is at least 9 blocks (our implementation min, iiuc)?

Reply:
Confusingly, the two limits are separate - there is a time limit on when you're allowed to use/send for the invoice, and then there is a CLTV limit on what the claim-timeout has to be when the invoice HTLC is received. This talks about the first, the 9 block limit talks about the second.